components/mailgun/sources/common-webhook.js

```diff
@@ -0,0 +1,92 @@
+const mailgun = require("../mailgun.app");
+
+module.exports = {
+  props: {
+    mailgun,
+    webhookSigningKey: {
+      type: "string",
+      secret: true,
+      label: "Mailgun webhook signing key",
+      description:
+<<<<<<< HEAD
+        "Your Mailgun webhook signing key, found [in your Mailgun dashboard](https://app.mailgun.com/app/dashboard), located under Settings on the left-hand nav and then in API Keys look for webhook signing key. Required to compute the authentication signature of events. ",
+=======
+        "Your Mailgun webhook signing key, found [in your Mailgun dashboard](https://app.mailgun.com/app/dashboard), located under Settings on the left-hand nav and then in API Keys look for webhook signing key. Required to compute the authentication signature of events.",
+>>>>>>> 3894ceaeca041938f51947f077049a80a007ebaf
+    },
+    http: "$.interface.http",
+    db: "$.service.db",
+  },
+  methods: {
+    verify(signingKey, timestamp, token, signature) {
+      const crypto = require("crypto");
+      const encodedToken = crypto
+        .createHmac("sha256", signingKey)
+        .update(timestamp.concat(token))
+        .digest("hex");
+      return encodedToken === signature;
+    },
+    emitEvent(eventWorkload) {
+      const eventPayload = eventWorkload["event-data"];
+      if (eventTypes.indexOf(eventPayload.event) > -1) {
```

Comment:
This is not defined anywhere.
Did you see any error logs while testing this?

BTW, there's a specific method for this case:
```suggestion
      if (eventTypes.includes(eventPayload.event)) {
```